### PR TITLE
Use mock to instead of TestShardingRule and TestTableRule in DataNodesTest class

### LINKED ISSUE #8067: Use mock to instead of TestShardingRule and TestTableRule

### PR DESCRIPTION
**Pull request by wenweibin**
Fixes #8067.

Changes proposed in this pull request:
- Use mock to instead of TestShardingRule and TestTableRule in DataNodesTest class.

### CODE REVIEW

File: shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/rule/DataNodesTest.java [hunk 1]
```diff
@@ -40,59 +39,60 @@
 import static org.mockito.Mockito.when;
 
 public final class DataNodesTest {
-
```
Comment (kimmking) [inline shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/rule/DataNodesTest.java:43]: keep four blank.

**Review by kimmking — CHANGES_REQUESTED** [review]

File: shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/rule/DataNodesTest.java [hunk 2]
```diff
@@ -79,18 +77,19 @@ public void assertGetDataNodeGroups() {
...
+        DataNodeContainedRule rule1 = mock(DataNodeContainedRule.class);
+        when(rule1.getAllDataNodes()).thenReturn(nodeMap);
         Map<String, Collection<String>> dataSourceMapper = Collections.singletonMap(logicDataSourceName, replicaDataSourceNames);
         DataSourceContainedRule rule2 = mock(DataSourceContainedRule.class);
         when(rule2.getDataSourceMapper()).thenReturn(dataSourceMapper);
         return new DataNodes(Arrays.asList(rule1, rule2));
     }
     
     private DataNodes getNonRoutedRuleDataNodes() {
-        return new DataNodes(Collections.singleton(new TestShardingSphereRule()));
```
Comment (tristaZero) [inline shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/rule/DataNodesTest.java:93]: Hi @wenweibin 
Do you think `TestShardingSphereRule` is no longer needed? If so, please delete them. Thanks.

**Review by tristaZero — CHANGES_REQUESTED** [review]

File: shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/rule/DataNodesTest.java (hunk 2 above)
Comment (wenweibin) [reply shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/rule/DataNodesTest.java:93]: Yes.I will delete them.

Comment (tristaZero) [reply shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/rule/DataNodesTest.java:93]: You will remove them in the next PR, won't you?

Comment (wenweibin) [reply shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/rule/DataNodesTest.java:93]: OK.I will finish it in next pr.

**Review by tristaZero — APPROVED** [review]
@wenweibin Nice, LGTM.

**Review by kimmking — APPROVED** [review]

**Pull request merged**